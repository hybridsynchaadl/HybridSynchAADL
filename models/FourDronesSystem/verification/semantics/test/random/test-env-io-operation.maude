
set show timing off .

load ../../interpreter-random.maude


----------------------------------------
mod TEST-ENV-FEATURE-OPERATION is
  inc ENV-FEATURE-OPERATION .
  pr CONC-REAL-EXP .
  ops a b c d e : -> FeatureId .
  ops cr1 cr2 : -> ComponentId .
endm

red readEnvFeature(
    < a :  EnvInPort | content : (real(0) # bool(true)),    envCache : (real(0) # bool(true)) >
    < b :  EnvInPort | content : (bool(true) # bool(true)), envCache : (bool(false) # bool(true))  >
    < c :  EnvInPort | content : (real(1) # bool(false)),   envCache : (real(2) # bool(false))    >
    < d : EnvOutPort | content : (real(0) # bool(false)) ,  envCache : (real(3) # bool(true)), target : cr1 >
    < e : EnvOutPort | content : (bool(true) # bool(true)), envCache : (real(4) # bool(false)), target : cr2 >) .


red writeEnvFeature(
    (e |-> (real(10000) # bool(true)))
,
    < a :  EnvInPort | content : (real(0) # bool(true)),    envCache : (real(0) # bool(true)) >
    < b :  EnvInPort | content : (bool(true) # bool(true)), envCache : (bool(false) # bool(true))  >
    < c :  EnvInPort | content : (real(1) # bool(false)),   envCache : (real(2) # bool(false))    >
    < d : EnvOutPort | content : (real(0) # bool(false)) ,  envCache : (real(3) # bool(true)), target : cr1 >
    < e : EnvOutPort | content : (bool(true) # bool(true)), envCache : (real(4) # bool(false)), target : cr2 >) .
