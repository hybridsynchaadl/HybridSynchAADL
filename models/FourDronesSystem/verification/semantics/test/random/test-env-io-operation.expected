==========================================
reduce in TEST-ENV-FEATURE-OPERATION : readEnvFeature((((< d : EnvOutPort |
    content : real(0) # bool(false),envCache : real(3) # bool(true),target :
    cr1 > < e : EnvOutPort | content : bool(true) # bool(true),envCache : real(
    4) # bool(false),target : cr2 >) < c : EnvInPort | content : real(1) #
    bool(false),envCache : real(2) # bool(false) >) < b : EnvInPort | content :
    bool(true) # bool(true),envCache : bool(false) # bool(true) >) < a :
    EnvInPort | content : real(0) # bool(true),envCache : real(0) # bool(true)
    >) .
rewrites: 48
result EnvFeatureReadResult: [< a : EnvInPort | content : [0.0] # [false],
    envCache : [0.0] # [true] > < b : EnvInPort | content : [false] # [false],
    envCache : [false] # [true] > < c : EnvInPort | content : [0.0] # [false],
    envCache : [2.0] # [false] > < d : EnvOutPort | content : [0.0] # [false],
    envCache : [3.0] # [true],target : cr1 > < e : EnvOutPort | content : [
    true] # [true],envCache : [4.0] # [false],target : cr2 >,a |-> ([0.0] # [
    true] : [true]), b |-> ([true] # [true] : [true]), c |-> ([1.0] # [false] :
    [true]), d |-> [0.0] # [false], e |-> [false] # [false]]
==========================================
reduce in TEST-ENV-FEATURE-OPERATION : writeEnvFeature(e |-> real(10000) #
    bool(true), (((< d : EnvOutPort | content : real(0) # bool(false),envCache
    : real(3) # bool(true),target : cr1 > < e : EnvOutPort | content : bool(
    true) # bool(true),envCache : real(4) # bool(false),target : cr2 >) < c :
    EnvInPort | content : real(1) # bool(false),envCache : real(2) # bool(
    false) >) < b : EnvInPort | content : bool(true) # bool(true),envCache :
    bool(false) # bool(true) >) < a : EnvInPort | content : real(0) # bool(
    true),envCache : real(0) # bool(true) >) .
rewrites: 22
result Configuration: < a : EnvInPort | content : [0.0] # [true],envCache : [
    0.0] # [true] > < b : EnvInPort | content : [true] # [true],envCache : [
    false] # [true] > < c : EnvInPort | content : [1.0] # [false],envCache : [
    2.0] # [false] > < d : EnvOutPort | content : [0.0] # [false],envCache : [
    3.0] # [true],target : cr1 > < e : EnvOutPort | content : [1.0e+4] # [
    true],envCache : [4.0] # [false],target : cr2 >
Bye.
